libs/variant/include/variant/variant.hpp

```diff
@@ -576,4 +584,167 @@ void Variant::IterateObject(Function const &function) const
 }
 
 }  // namespace variant
+
+namespace serializers {
+
+template <typename D>
+struct ForwardSerializer<fetch::variant::Variant, D>
+{
+public:
+  using Type       = fetch::variant::Variant;
+  using DriverType = D;
+
+  template <typename Serializer>
+  static void Serialize(Serializer &serializer, Type const &var)
+  {
+    auto typecode = static_cast<int>(var.type());
+    serializer << typecode;
+    switch (var.type())
+    {
+    case Type::Type::UNDEFINED:
+    {
+      return;
+    }
+
+    case Type::Type::NULL_VALUE:
+    {
+      return;
+    }
+    case Type::Type::INTEGER:
+    {
+      serializer << var.As<int>();
+      return;
+    }
+    case Type::Type::FLOATING_POINT:
+    {
+      serializer << var.As<double>();
+      return;
+    }
+    case Type::Type::FIXED_POINT:
+    {
+      serializer << var.As<int>();
+      return;
+    }
+    case Type::Type::BOOLEAN:
+    {
+      serializer << var.As<bool>();
+      return;
+    }
+    case Type::Type::STRING:
+    {
+      serializer << var.As<std::string>();
```

Comment:
std::string will work, but it would be better to use byte_array::ConstByteArray (no performance hit caused by conversion string -> byte aray)

Reply:
done!